tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -465,6 +465,116 @@ TEST(LinalgBackendEigen, SGMatrix_cross_entropy)
 	EXPECT_NEAR(ref, result, 1e-15);
 }
 
+TEST(LinalgBackendEigen, SGMatrix_pinv_psd)
```

Comment:
name it pinvh after the method

and then split this into two tests, one for each method. You could use a fixture to share the data

Reply:
i take this back. lets just leave it like this